lib/source-map/source-node.js

```diff
@@ -375,8 +373,8 @@ define(function (require, exports, module) {
         lastOriginalSource = null;
         sourceMappingActive = false;
       }
-      chunk.match(REGEX_CHARACTER).forEach(function (ch, idx, array) {
-        if (REGEX_NEWLINE.test(ch)) {
+      chunk.replace(REGEX_WIN_NEWLINE, "\n").split('').forEach(function (ch, idx, array) {
+        if (ch === "\r\n" || ch === "\n") {
```

Comment:
This check should just be `if (ch === "\n")` now. Also, further down you can replace `+= ch.length` with `+= 1` (or `++`.)